include/swift/AST/DiagnosticsSema.def

```diff
@@ -547,6 +547,9 @@ ERROR(invalid_member_type,none,
 ERROR(invalid_member_type_suggest,none,
       "%0 does not have a member type named %1; did you mean %2?",
       (Type, Identifier, Identifier))
+ERROR(invalid_member_reference,none,
+      "%0 %1 is not a member type of %2",
+      (DescriptiveDeclKind, Identifier, Type))
```

Comment:
I think "nested type" might work better here than "member type", but I don't have a strong opinion about it.